File: fonte/api/src/main/java/com/AchadosPerdidos/API/Domain/Entity/Endereco.java
package com.AchadosPerdidos.API.Domain.Entity;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Endereco {
    private Integer id;
    private String logradouro;
    private String numero;
    private String complemento;
    private String bairro;
    private String cep;
    private Integer cidadeId;
    private Date dtaCriacao;
    private Boolean flgInativo;
    private Date dtaRemocao;
}

File: flutter/lib/domain/entities/endereco.dart
class Endereco {
  final int id;
  final String logradouro;
  final String? numero;
  final String? complemento;
  final String? bairro;
  final String? cep;
  final int cidadeId;
  final String? cidadeNome;
  final String? estadoUf;
  final DateTime? dtaCriacao;
  final bool flgInativo;
  final DateTime? dtaRemocao;

  Endereco({
    required this.id,
    required this.logradouro,
    this.numero,
    this.complemento,
    this.bairro,
    this.cep,
    required this.cidadeId,
    this.cidadeNome,
    this.estadoUf,
    this.dtaCriacao,
    this.flgInativo = false,
    this.dtaRemocao,
  });
}

